cli/js/compiler.ts

```diff
@@ -240,6 +240,12 @@ class SourceFile {
   }
 }
 
+const TS_CHECK_PRAGMA_RE = /^\/\/\s?@ts-check/;
+
+function hasTsCheckPragma(sourceCode: string): boolean {
+  return TS_CHECK_PRAGMA_RE.test(sourceCode);
+}
```

Comment:
I was thinking about this...  The problem is that both Rust and the compiler need to know if it is there, as they have different behaviours.

We should detect this once in Rust and pass it to TypeScript as part of the SourceFile structure (e.g. `"tsCheck": true`) when the file is retrieved.  That way we only have to maintain one regex and only do a single extra scan of JS files.

Reply:
Agree with @kitsonk, that said I still don't see a way to implement this function as described [in this comment](https://github.com/denoland/deno/pull/3114#issuecomment-541342649). There's no way to do that after loading some files into isolate - they should be preprocessed beforehand.

Reply:
What you send to the compiler, and what you load in the isolate are two different things.  Those already loaded into the isolate will simply not actually be written out by the compiler anyways, but once you hit the first `@ts-check`, spin up the compiler with the original root names, it will end up fetching everything else and only writing out those with `ts-check`.

Reply:
@kitsonk I get what we should do, but our `Loader` API doesn't play nicely with this idea.
`State::load` has only module specifier for current module (no referrer); granted I could take root module from `main_module`, but... during dynamic import situation would be the same and then there's no way to get the "root" of dynamic import. 

I do agree this is a proper way to solve this PR, I just haven't established how to do that. I was thinking about preprocessing files because it's way easier than tweaking `Loader`, but surely it might be better in the long run to actually tweak `Loader`. 

Anyway, I'm focusing on ops ATM and we'll get back to this PR afterwards. So if you want to continue this PR feel free!